### ISSUE TITLE
Add documentation for levels

### ISSUE THREAD

**Issue by dreamflasher** [Iotic-Labs/py-lz4framed#5]
At https://github.com/lz4/lz4 is a benchmark for different lz4 settings. I am interested in maximum decompression throughput, currently the best for that is "LZ4 fast 8 (v1.7.3)". How do I specify this setting with py-lz4framed?

**Comment by vtermanis** [comment]
There are some compression-level constants [exposed directly from lz4](https://github.com/Iotic-Labs/py-lz4framed/blob/master/lz4framed/py-lz4framed.c#L1042), so you could do something like:
```python
for level in range(lz4framed.LZ4F_COMPRESSION_MIN, lz4framed.LZ4F_COMPRESSION_MAX + 1):
  lz4framed.compress(b'some data', level=level)
```
I don't know which one of those corresponds to "fast 8" - it's got to say that somewhere in the lz4 documentation (or otherwise you could ask there). Maybe it is just 8 (see e.g. [here - anything below 9 maybe](https://github.com/lz4/lz4/blob/ca083dd4daac795600a9d7be56e0735d0076b600/programs/lz4cli.c#L118)).

But basically all constants visible through py-lz4framed come straight from the lz4 library.

**Edit**: Update code example

**Comment by vtermanis** [comment]
See also `help(lz4framed.compress)` and https://github.com/Iotic-Labs/py-lz4framed/blob/master/lz4framed/__init__.py#L78

**Comment by dreamflasher** [comment]
Thanks, http://python-lz4.readthedocs.io/en/latest/lz4.block.html exposes `lz4.block.compress(source, mode='default', acceleration=1, compression=0)` with default/fast/high_compression mode and acceleration/compression, so I guessed it could be fast with acceleration 8. But maybe they don't provide this in the latest version anymore?

**Comment by vtermanis** [comment]
Ah ok, I see now what you mean.

So the fast call options are only available through the [block API](https://github.com/lz4/lz4/blob/v1.7.5/lib/lz4.h#L170). This project only supports the frame API (because this means it's [interoperable](http://lz4.github.io/lz4/#interoperable-lz4) whereas if you just compress raw blocks you need to decide e.g. how to transmit the length etc. yourself.) I guess you could use the [python-lz4](https://github.com/python-lz4/python-lz4) module if you want to use the acceleration parameters (but as you can [see here](http://python-lz4.readthedocs.io/en/latest/lz4.frame.html#lz4.frame.compress) the frame API also doesn't expose acceleration). I can't comment on how stable that project is given I've never used it.

For the frame API the compression level is set only [here](https://github.com/lz4/lz4/blob/dev/lib/lz4frame.h#L167) (note no acceleration or other parameters). You could ask the [lz4](https://github.com/lz4/lz4) project whether the frame API intends to support the accelerated parameter in the near future.

From what I can tell anything under `lz4framed.LZ4F_COMPRESSION_MIN_HC` (or [here in C API](https://github.com/lz4/lz4/blob/8c4de60d0f871f8e81986b2c61b70e1bcb8b97fd/lib/lz4hc.h#L47)) counts as "fast".

See also e.g. [this gist](https://gist.github.com/vtermanis/ddb42d56dcc8110623938ff4a643e09f) from a while back where output side can be the same for low levels. (Obviously that test case is nonsense because the data is very synthetic.)

**Comment by dreamflasher** [comment]
Thank you very much, that explains a lot and solves this issue! Thank you!

**Comment by vtermanis** [comment]
@DreamFlasher  - Once [this](https://github.com/lz4/lz4/commit/e2c9b19122a19fa8a46a253e4ce556a5b75fa8b1#diff-139f9c8285cef1fcd02b6a72cc74241dR170) is released (v1.7.6 of lz4 I think), accelerated mode will be possible. (No idea when lz4 plan to release though.)

**Comment by vtermanis** [comment]
For anyone else who'd like to use acceleration, new lz4 (with acceleration support) has no planned release for now. If you're interested in using it in `lz4framed`, post [here](https://github.com/lz4/lz4/issues/346) to express interest.

**Comment by vtermanis** [comment]
Acceleration is now available (since `v0.1.0`)
